edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/ProxyReadinessProbe.cs

```diff
@@ -25,8 +25,10 @@ async Task<ProxyReadiness> CheckAsync(CancellationToken token)
         {
             try
             {
-                HttpResponseMessage response = await this.client.GetAsync($"/systeminfo?api-version={this.apiVersion}", token);
-                return response.StatusCode == HttpStatusCode.OK ? ProxyReadiness.Ready : ProxyReadiness.Failed;
+                using (HttpResponseMessage response = await this.client.GetAsync($"/systeminfo?api-version={this.apiVersion}", token))
```

Comment:
I feel all of these should be able to be caught with some sort of code analytics thing... with some warning like "WRN: response resource not being handled..." Do we have something like that already? If so, are we not running it? If not, should we add something like that?

Reply:
I don't have answer to that. Quick search showed this - https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000 - but this is not applicable to `System.IO.Stream`.

Reply:
Yeah not sure on what the tool might be, but I will look into this a bit more... as maybe we can further automate?